apollo-client/src/main/java/com/ctrip/framework/apollo/internals/ConfigMonitorInitializer.java

```diff
@@ -0,0 +1,129 @@
+
+import com.ctrip.framework.apollo.build.ApolloInjector;
+import com.ctrip.framework.apollo.core.utils.ClassLoaderUtil;
+import com.ctrip.framework.apollo.monitor.internal.exporter.AbstractApolloClientMetricsExporter;
+import com.ctrip.framework.apollo.monitor.internal.exporter.ApolloClientMetricsExporter;
+import com.ctrip.framework.apollo.monitor.internal.jmx.ApolloClientJmxMBeanRegister;
+import com.ctrip.framework.apollo.monitor.internal.ApolloClientMonitorContext;
+import com.ctrip.framework.apollo.monitor.internal.listener.impl.DefaultApolloClientBootstrapArgsApi;
+import com.ctrip.framework.apollo.monitor.internal.listener.impl.DefaultApolloClientExceptionApi;
+import com.ctrip.framework.apollo.monitor.internal.listener.impl.DefaultApolloClientNamespaceApi;
+import com.ctrip.framework.apollo.monitor.internal.listener.impl.DefaultApolloClientThreadPoolApi;
+import com.ctrip.framework.apollo.monitor.internal.exporter.ApolloClientMetricsExporterFactory;
+import com.ctrip.framework.apollo.monitor.internal.tracer.ApolloClientMonitorMessageProducer;
+import com.ctrip.framework.apollo.monitor.internal.tracer.ApolloClientMessageProducerComposite;
+import com.ctrip.framework.apollo.tracer.internals.NullMessageProducer;
+import com.ctrip.framework.apollo.tracer.internals.cat.CatMessageProducer;
+import com.ctrip.framework.apollo.tracer.internals.cat.CatNames;
+import com.ctrip.framework.apollo.tracer.spi.MessageProducer;
+import com.ctrip.framework.apollo.util.ConfigUtil;
+import com.ctrip.framework.foundation.internals.ServiceBootstrap;
+import java.util.List;
+
+/**
+ * ConfigMonitorInitializer initializes the Apollo Config Monitor.
+ */
+public class ConfigMonitorInitializer {
+
+  protected static boolean hasInitialized = false;
+  private static ConfigUtil m_configUtil = ApolloInjector.getInstance(ConfigUtil.class);
+  private static ApolloClientMonitorContext monitorContext = ApolloInjector.getInstance(
+      ApolloClientMonitorContext.class);
+
+  public static void initialize() {
+    if (m_configUtil.isClientMonitorEnabled() && !hasInitialized) {
+      synchronized (ConfigMonitorInitializer.class) {
+        if (!hasInitialized) {
+          doInit();
+          hasInitialized = true;
+        }
+      }
+    }
+  }
+
+  private static void doInit() {
+    initializeMetricsEventListener();
+    initializeMetricsExporter();
+    initializeJmxMonitoring();
+    hasInitialized = true;
+  }
+
+
+  private static void initializeJmxMonitoring() {
+    if (m_configUtil.isClientMonitorJmxEnabled()) {
+      monitorContext.getCollectors().forEach(metricsCollector ->
+          ApolloClientJmxMBeanRegister.register(
+              MBEAN_NAME + metricsCollector.getName(), metricsCollector)
+      );
+    }
+  }
+
+  private static void initializeMetricsEventListener() {
+    DefaultConfigManager configManager = (DefaultConfigManager) ApolloInjector.getInstance(
```

Comment:
This will cause class cast exception if user use their own ConfigManager instead of DefaultConfigManager.

We can add some default implement to interface ConfigManager, then use ConfigManager directly